modules/amxIdSystem.js

```diff
@@ -0,0 +1,150 @@
+  if (config == null || config.storage == null) {
+    logError(`${NAME}: config.storage is required.`);
+    return false;
+  }
+
+  if (config.storage.type !== 'html5') {
+    logError(
+      `${NAME} only supports storage.type "html5". ${config.storage.type} was provided`
+    );
+    return false;
+  }
+
+  if (
+    typeof config.storage.expires === 'number' &&
+    config.storage.expires > 30
+  ) {
+    logError(
+      `${NAME}: storage.expires must be <= 30. ${config.storage.expires} was provided`
+    );
+    return false;
+  }
+
+  return true;
+}
+
+function handleSyncResponse(client, response, callback) {
+  if (response.id != null && response.id.length > 0) {
+    callback(response.id);
+    return;
+  }
+
+  if (response.u == null || response.u.length === 0) {
+    callback(null);
+    return;
+  }
+
+  client(response.u, {
+    error(e) {
+      logError(`${NAME} failed on ${response.u}`, e);
+      callback(null);
+    },
+    success(complete) {
+      if (complete != null && complete.length > 0) {
+        const value = JSON.parse(complete);
+        if (value.id != null) {
+          callback(value.id);
+          return;
+        }
+      }
+
+      logError(`${NAME} invalid value`, complete);
+      callback(null);
+    },
+  });
+}
+
+export const amxIdSubmodule = {
+  /**
+   * @type {string}
+   */
+  name: NAME,
+
+  /**
+   * @type {string}
+   */
+  version,
+
+  /**
+   * IAB TCF Vendor ID
+   * @type {string}
+   */
+  gvlid: GVL_ID,
+
+  decode: (value) =>
+    value != null && value.length > 0
+      ? { [ID_KEY]: value }
+      : undefined,
+
+  getId(config, consentData, _extant) {
+    if (!validateConfig(config)) {
+      return undefined;
+    }
+
+    const consent = consentData || { gdprApplies: false, consentString: '' };
+    const timeout = deepAccess(config, 'params.timeout', 300);
+    const client = ajaxBuilder(timeout);
+    const usp = uspDataHandler.getConsentData();
+    const ref = getRefererInfo();
+
+    const params = {
+      tagId: deepAccess(config, 'params.tagId', ''),
+      ref: ref.referer,
+      u: ref.stack[0] || getWindowTop().location.href,
+      v: '$prebid.version$',
+      vg: '$$PREBID_GLOBAL$$',
+      us_privacy: usp,
+      gdpr: consent.gdprApplies ? 1 : 0,
+      gdpr_consent: consent.consentString,
+    };
+
+    const callback = (done) =>
+      client(
+        SYNC_URL,
+        {
+          error(e) {
+            logError(`${NAME} failed to load`, e);
+            done(null);
+          },
+          success(responseText) {
+            if (responseText != null && responseText.length > 0) {
+              try {
+                const parsed = JSON.parse(responseText);
+                handleSyncResponse(client, parsed, done);
+                return;
+              } catch (e) {
+                logError(`${NAME} invalid response`, responseText);
+              }
+            }
+
+            done(null);
+          },
+        },
+        params
```

Comment:
when i add run your code on page (I'd recommend adding it to `userId_example.html` under integration examples), i get an error with the response from the server. i believe `params` should be `JSON.stringify(parms)` here as it shows as `[object Object]` in the network call. either that, or the server is not responding properly.